api/circulation.py

```diff
@@ -171,6 +171,11 @@ class CirculationAPI(object):
     def __init__(self, library, api_map=None):
         """Constructor.
 
+        # TODO: This needs to take a database connection as well so
+        # that we can keep the scoped session. It should only store
+        # library_id or patron_id. OR, we need to make sure we create
+        # a new CirculationAPI on each request.
+
```

Comment:
If we don't fix this, we'll have data integrity problems when we run in a multi-threaded environment.